src/server.h

```diff
@@ -733,7 +733,8 @@ typedef struct RedisModuleDigest {
 #define OBJ_STATIC_REFCOUNT (INT_MAX-1) /* Object allocated in the stack. */
 #define OBJ_FIRST_SPECIAL_REFCOUNT OBJ_STATIC_REFCOUNT
 typedef struct redisObject {
-    unsigned type:4;
+    unsigned hasexpire:1;
+    unsigned type:3;
```

Comment:
OK, that's fine by me, but there is no guarantee that `int` is 32 bits, so if we change like this:

```diff
-int refcount;
+unsigned hasexpire:1;
+unsigned refcount:31;
```

... then we shouldn't use INT_MAX for the highest refcount value, but instead we should use 0x7fffffff explicitly, the maximum 31 bit value.

It would also be very easy to change this later, if we need more types.